src/init/features/hosting/index.ts

```diff
@@ -29,19 +33,124 @@ export async function doSetup(setup: any, config: any): Promise<void> {
   logger.info("have a build process for your assets, use your build's output directory.");
   logger.info();
 
+  let discoveredFramework = previews.frameworkawareness ?
+    await discover(config.projectDir, false) :
+    undefined;
+
+  if (previews.frameworkawareness) {
+
+    if (discoveredFramework) await prompt(setup.hosting, [
+      {
+        name: "useDiscoveredFramework",
+        type: "confirm",
+        default: true,
+        message: `Detected an existing ${WebFrameworks[discoveredFramework.framework].name} codebase in the current directory, should we use this?`,
+      },
+    ]);
```

Comment:
Rather nasty one-liner. Could you prefer to at least put the `await prompt` on a new line, if not in a bracketed block?